File: web/src/types/course.ts
export type CourseStatus = 'initializing' | 'learning' | 'paused' | 'completed'
export type GenerationStatus = 'not_started' | 'in_progress' | 'ready'
export type LearningStatus = 'not_started' | 'in_progress' | 'paused' | 'completed'
export type MasteryStatus = 'unseen' | 'developing' | 'stable' | 'needs_review'

export interface Course {
  id: number
  name: string
  description: string
  goal: string
  status: CourseStatus
  /** @deprecated compatibility only; use the three explicit progress fields. */
  progress: number
  generation_status: GenerationStatus
  generation_progress: number
  learning_status: LearningStatus
  coverage_progress: number
  mastery_status: MasteryStatus
  mastery_progress: number
  current_unit: string
  current_unit_id?: number | null
  current_lesson_id?: number | null
  last_studied_at: string | null
  created_at: string
  updated_at: string
}

File: internal/model/course.go
package model

import "time"

type CourseStatus string

const (
	CourseStatusInitializing CourseStatus = "initializing"
	CourseStatusLearning     CourseStatus = "learning"
	CourseStatusPaused       CourseStatus = "paused"
	CourseStatusCompleted    CourseStatus = "completed"
)

type Course struct {
	ID              uint         `json:"id" gorm:"primaryKey"`
	UserID          uint         `json:"user_id" gorm:"not null;index"`
	Name            string       `json:"name" gorm:"size:120;not null"`
	Description     string       `json:"description" gorm:"type:text"`
	Goal            string       `json:"goal" gorm:"type:text"`
	Status          CourseStatus `json:"status" gorm:"size:32;not null;index"`
	Progress        int          `json:"progress" gorm:"not null;default:0"`
	CurrentUnit     string       `json:"current_unit" gorm:"size:160"`
	CurrentUnitID   *uint        `json:"current_unit_id" gorm:"index"`
	CurrentLessonID *uint        `json:"current_lesson_id" gorm:"index"`
	LastStudiedAt   *time.Time   `json:"last_studied_at"`
	CreatedAt       time.Time    `json:"created_at"`
	UpdatedAt       time.Time    `json:"updated_at"`
}
